src/System.Diagnostics.TraceSource/src/System/Diagnostics/DefaultTraceListener.cs

```diff
@@ -115,7 +115,34 @@ private void WriteAssert(string stackTrace, string message, string detailMessage
         /// </devdoc>
         public override void Write(string message)
         {
-            if (NeedIndent) WriteIndent();
+            Write(message, true);
+        }
+
+        /// <devdoc>
+        ///    <para>
+        ///       Writes the output followed by a line terminator using <see cref="System.Diagnostics.Debug.Write"/>.
+        ///    </para>
+        /// </devdoc>
+        public override void WriteLine(string message)
+        {
+            WriteLine(message, true);
+        }
+
+        private void WriteLine(string message, bool useLogFile)
+        {
+            if (NeedIndent) 
+                WriteIndent();
+
+            // I do the concat here to make sure it goes as one call to the output.
+            // we would save a stringbuilder operation by calling Write twice.
```

Comment:
Nit: I realize you just copied this comment from the original, it could be cleaned up slightly, though, e.g.
```C#
// The concat is done here to enable a single call to Write
```
I have a small pet peeve against of "I" in shared codebases.